### PR TITLE
Fix conflicting enum to_py function with multiple modules

### LINKED ISSUE #5860: [BUG] migration to cython3.0.5: redefinition of '__Pyx_Enum_<enum_name>_to_py
### Describe the bug

I'm having an issue with migrating PyVerbs package, part of the upstream rdma-core, to cython3.
We've been working for a few years with Cython 0.2X.YY versions without the below issues.
And I didn't find a solved issue for this or any relevant notes in the [migration document](https://docs.cython.org/en/latest/src/userguide/migrating_to_cy30.htm).

```
<path-to-rdma-core>/rdma-core/build-fc39/pyverbs/srq.c:5528:18: error: redefinition of '__Pyx_Enum_ibv_ops_wr_opcode_to_py'
5528 | static PyObject *__Pyx_Enum_ibv_ops_wr_opcode_to_py(enum ibv_ops_wr_opcode __pyx_v_c_val) {
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<path-to-rdma-core>/rdma-core/build-fc39/pyverbs/srq.c:4511:18: note: previous definition of '__Pyx_Enum_ibv_ops_wr_opcode_to_py' with type 'PyObject *(enum ibv_ops_wr_opcode)' {aka 'struct _object *(enum ibv_ops_wr_opcode)'}
4511 | static PyObject *__Pyx_Enum_ibv_ops_wr_opcode_to_py(enum ibv_ops_wr_opcode __pyx_v_c_val) {
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[420/575] Building C object pyverbs/CMakeFiles/cq.cpython-312-x86_64-linux-gnu.dir/cq.c.o
```

### Code to reproduce the behaviour:

Go to [rdma-core](https://github.com/linux-rdma/rdma-core) project and checkout this PR branch:[Update to fc39]( https://github.com/linux-rdma/rdma-core/pull/1405)
Then, to build, and reproduce, you can do:
```
buildlib/cbuild build-images fc39
buildlib/cbuild make fc39
```

### Expected behaviour

I expect the compilation to pass as it's been passing on any Cython 0.29.XY versions.
The cythonization process creates a duplicate enum definition in the C files due to the "inlucde <pxd_file_name>.pxd" lines.

### OS

Linux

### Python version

3.10.8

### Cython version

3.0.5 (actually from any cython3 version)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by da-woods**
Fix conflicting names of cpdef enum to_py functions when the an enum with the name name exists in multiple modules. Instead use the cname to name the to_py function since we have already ensured that it is unique and mangled with the module name.

Possibly fixes #5860 (it definitely fixes a real bug, but that project has far too many dependencies for me to test, so who knows if it fixes *that* bug).

### CODE REVIEW

File: tests/run/cpdef_enums_import.srctree [hunk 1]
```diff
@@ -37,6 +45,7 @@ cpdef enum EnumTypeNotInPyx:
 
 from enums cimport *
 from enums_without_pyx cimport *
+import enums2
```
Comment (scoder) [inline tests/run/cpdef_enums_import.srctree:48]:
```suggestion
cimport enums2
```

File: tests/run/cpdef_enums_import.srctree [hunk 2]
```diff
@@ -28,6 +28,14 @@ cpdef enum NamedEnumType:
 
 cpdef foo()
 
+######## enums2.pyx ############
+
+######## enums2.pxd ############
```
Comment (scoder) [inline tests/run/cpdef_enums_import.srctree:33]:
```suggestion
######## enums_same_name.pyx ############

######## enums_same_name.pxd ############
```

**Review by scoder — APPROVED** [review]
Good catch.

**Comment by da-woods** [comment]
3.0.x commit e6490a642

**Pull request merged**